grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -1137,6 +1141,7 @@ class ProductService {
 
         } catch (Exception e) {
             log.warn("Error generating unique product code " + e.message, e)
+            throw e
```

Comment:
I'd remove this `try/catch` at all, since it does only `log.warn` and it prevents our errors handler from working if you don't do `try/catch` also in controller as you did above.